lib/puppet/type/file/content.rb

```diff
@@ -220,6 +221,19 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
```

Comment:
This is the equivalent to the pre-existing `get_from_source` (now `get_from_puppet_source`) above, which will bypass the indirector for `puppet://` URLs.

The reason for this is optimizing the handling of large files (shoutout to @masterzen for pointing this out to me in the old PR). These methods here take a block and stream the contents. The indirector, on the other hand, will store the whole file content in a variable and return it for processing. This implies considerable memory consumption for large files.